### ISSUE TITLE
`high_resolution_clock` is discouraged, use `steady_clock` instead

### ISSUE THREAD

**Issue by PhilMiller** [kokkos/kokkos#5584]
Separate from this PR, `high_resolution_clock` is not exactly deprecated, but its use is discouraged by knowledgeable folks: https://en.cppreference.com/w/cpp/chrono/high_resolution_clock

For this use case, using `steady_clock` directly is currently preferred

High quality software chronometry is unduly hard

_Originally posted in https://github.com/kokkos/kokkos/pull/5577#discussion_r999945688_

**Comment by ajpowelsnl** [comment]
Shoud `high_resolution_clock` officially be deprecated in Kokkos-4.0?

**Comment by dalg24** [comment]
No, [`std:: high_resolution_clock`](https://en.cppreference.com/w/cpp/chrono/high_resolution_clock) belongs to the standard library.  Phil is saying we suggesting we avoid it in Kokkos.  See the **Notes** section if you follow the link to the documentation.
